src/GovUk.Education.ExploreEducationStatistics.Data.Api/Services/PermalinkService.cs

```diff
@@ -63,27 +65,124 @@ public class PermalinkService : IPermalinkService
+                            cancellationToken: cancellationToken);
+
+                        permalink.Table = JsonConvert.DeserializeObject<dynamic>(tableJson);
+                        return permalink;
+                    }
+                    catch (FileNotFoundException)
+                    {
+                        return new NotFoundResult();
+                    }
+                });
+        }
+
+        public async Task<Either<ActionResult, PermalinkViewModel>> CreatePermalink(PermalinkCreateRequest request,
+            CancellationToken cancellationToken = default)
+        {
+            return await _subjectRepository.FindPublicationIdForSubject(request.Query.SubjectId)
+                .OrNotFound()
+                .OnSuccess(publicationId => _releaseRepository.GetLatestPublishedRelease(publicationId))
+                .OnSuccess(release => CreatePermalink(release.Id, request, cancellationToken));
+        }
+
+        public async Task<Either<ActionResult, PermalinkViewModel>> CreatePermalink(Guid releaseId,
+            PermalinkCreateRequest request,
+            CancellationToken cancellationToken = default)
+        {
+            return await _tableBuilderService.Query(releaseId, request.Query, cancellationToken)
+                .OnSuccessCombineWith(tableResult => _frontendService.CreateUniversalTable(tableResult,
+                    request.Configuration,
+                    cancellationToken))
+                .OnSuccessCombineWith(async tuple =>
+                {
+                    var (tableResult, _) = tuple;
+                    return await _permalinkCsvMetaService.GetCsvMeta(request.Query.SubjectId,
+                        tableResult.SubjectMeta.Locations,
+                        tableResult.SubjectMeta.Filters,
+                        tableResult.SubjectMeta.Indicators,
+                        cancellationToken);
+                })
+                .OnSuccess(async tuple =>
+                {
+                    var (tableResult, universalTable, csvMeta) = tuple;
+                    var subjectMeta = tableResult.SubjectMeta;
+                    var permalink = new Permalink
+                    {
+                        ReleaseId = releaseId,
+                        SubjectId = request.Query.SubjectId,
+                        PublicationTitle = subjectMeta.PublicationName,
+                        DataSetTitle = subjectMeta.SubjectName,
+                        CountFilterItems = CountFilterItems(subjectMeta.Filters),
+                        CountFootnotes = subjectMeta.Footnotes.Count,
+                        CountIndicators = subjectMeta.Indicators.Count,
+                        CountLocations = CountLocations(subjectMeta.Locations),
+                        CountObservations = tableResult.Results.Count(),
+                        CountTimePeriods = subjectMeta.TimePeriodRange.Count,
+                        Table = universalTable
```

Comment:
We're storing the entire table in the database as well as blob storage? Is this just to help with the migration or are we planning on keeping it like this?

Seems like there's potentially issues here as we're redundantly storing it in two places which seems overkill. If we feel like having the tables in the content DB is beneficial for some reason (like reporting/querying), then perhaps we should just ditch storing them in blob storage completely.

However, there might be some storage size and performance considerations for putting these in the content DB we should consider. If there isn't a compelling reason to move over to the content DB, then would suggest we just keep them blob storage.

Reply:
Ah, I see the confusion here. `Permalink.Table` is not mapped to the database: 🙂 

```
    [NotMapped]
    public dynamic? Table { get; set; }
```

We are not storing the entire table in the database as well as blob storage and the intention has never been to do so, so there's no issue with redundancy or size and performance concerns.

It was just written like that so that the `PermalinkService.Find` method could be written a little nicer and return a `Permalink` that also has the table content fetched from blob storage.

I can look at refactoring that to make it a little less confusing.

Reply:
Cool, did miss that part! However, it seems like a potential source of confusion (as this has demonstrated), so would suggest that we move it off the `Permalink` model and just pass the table through as a separate argument to any relevant methods.